messageregistry_test.go

```diff
@@ -0,0 +1,370 @@
+package dogma_test
+
+import (
+	"reflect"
+	"strings"
+	"testing"
+
+	. "github.com/dogmatiq/dogma"
+)
+
+func TestRegisteredMessageTypeFor(t *testing.T) {
+	t.Run("it returns the type that represents T", func(t *testing.T) {
+		const id = "2d8ce56f-1983-44e3-a55d-f74d8dcb0adc"
+		type T struct{ Command }
+		RegisterCommand[T](id)
+
+		mt, ok := RegisteredMessageTypeFor[T]()
+		if !ok {
+			t.Fatal("expected message type to be registered")
+		}
+
+		if got, want := mt.GoType(), reflect.TypeFor[T](); got != want {
+			t.Fatalf("unexpected type: got %v, want %v", got, want)
+		}
+	})
+
+	t.Run("it returns false when T is not in the registry", func(t *testing.T) {
+		type T struct{ Command }
+
+		_, ok := RegisteredMessageTypeFor[T]()
+		if ok {
+			t.Fatal("did not expect message type to be registered")
+		}
+	})
+}
+
+func TestRegisteredMessageTypeByID(t *testing.T) {
+	t.Run("it returns the type associated with the normalized ID", func(t *testing.T) {
+		const id = "37264B0D-4342-4708-8263-60D82DE78AD1"
+		type T struct{ Event }
+		RegisterEvent[T](id)
+
+		mt, ok := RegisteredMessageTypeByID(id)
+		if !ok {
+			t.Fatal("expected message type to be registered")
+		}
+
+		if got, want := mt.GoType(), reflect.TypeOf(T{}); got != want {
+			t.Fatalf("unexpected type: got %v, want %v", got, want)
+		}
+	})
+
+	t.Run("it returns false when ID is not registered", func(t *testing.T) {
+		const id = "75285aae-f85a-435b-ad36-7471ab169348"
+
+		_, ok := RegisteredMessageTypeByID(id)
+		if ok {
+			t.Fatal("did not expect message type to be registered")
+		}
+	})
+
+	t.Run("panics when the ID is invalid", func(t *testing.T) {
+		expectPanic(
+			t,
+			`"<non-uuid>" is not a canonical RFC 4122 UUID: expected 36 characters`,
+			func() {
+				RegisteredMessageTypeByID("<non-uuid>")
+			},
+		)
+	})
```

Comment:
Nit picky but defining invalid UUID
```suggestion
		const id = "75285aae-f85a-435b-ad36-7471ab169348"
		const invalidUUID = "<non-uuid>"

		_, ok := RegisteredMessageTypeByID(id)
		if ok {
			t.Fatal("did not expect message type to be registered")
		}
	})

	t.Run("panics when the ID is invalid", func(t *testing.T) {
		expectPanic(
			t,
			`"<non-uuid>" is not a canonical RFC 4122 UUID: expected 36 characters`,
			func() {
				RegisteredMessageTypeByID(invalidUUID)
			},
		)
	})
```